Node.js/samplerequest.js

```diff
@@ -71,15 +71,21 @@ mqBoilerPlate.initialise('PUT')
     debug_info('Getting Response for ', msgID);
     return mqBoilerPlate.getMessagesDynamicQueue(msgID, msgCB);
   })
+  .then(() => { // We kickstart the async process to listen for any messages on the Reply to Queue.
+    debug_info('Kick start the get callback');
+    return mqBoilerPlate.startGetAsyncProcess();
+  })  
   .then(() => {
     debug_info('Waiting for termination');
     return mqBoilerPlate.checkForTermination();
   })
   .then(() => {
     mqBoilerPlate.teardown();
+    debug_info('Application Completed');
```

Comment:
Like this, this async anonymous function returns a promise. It might be better to return the teardown, and have a subsequent then that runs the process.exit if needed.